dispatch/apps/api/views.py

```diff
@@ -638,24 +638,35 @@ def list(self, request):
 #     except:
 #         return Response(request.data, status=status.HTTP_400_BAD_REQUEST)
 
-
-@api_view(['POST'])
 @permission_classes((AllowAny,))
-def user_authenticate(request):
+class AuthenticationViewSet(viewsets.GenericViewSet):
 
-    email = request.data.get('email', None)
-    password = request.data.get('password', None)
+    def user_authenticate(self, request):
+        email = request.data.get('email', None)
+        password = request.data.get('password', None)
 
-    user = authenticate(username=email, password=password)
+        user = authenticate(username=email, password=password)
 
-    if user is not None and user.is_active:
+        if user is not None and user.is_active:
 
-        (token, created) = Token.objects.get_or_create(user=user)
+            (token, created) = Token.objects.get_or_create(user=user)
 
-        data = {
-            'token': unicode(token)
-        }
+            data = {
+                'token': unicode(token)
+            }
 
-        return Response(data, status=status.HTTP_202_ACCEPTED)
-    else:
-        return Response({}, status=status.HTTP_400_BAD_REQUEST)
+            return Response(data, status=status.HTTP_202_ACCEPTED)
+        else:
+            return Response({}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Recognising that this code was already here, so feel free to ignore:

- Do you think 403 would be a more semantic and standard response here? 400 generally corresponds to "you've sent malformed or bad data", whereas the case here seems more like the request was understood but the server is unwilling to accept it -- 403.
  - https://tools.ietf.org/html/rfc7231#section-6.5.1

- 400 would be appropriate if `email` or `password` or both were not sent at all. On that note, are there going to be cases where attempting to login with just an email or just a password is ok? If not, it would be better for lines 645-646 to look more like:
    ```py
    try:
        email = request.data.get('email')
        password = request.data.get('password')
    except KeyError:
        return Response({}, status=status.HTTP_400_BAD_REQUEST)